### PR TITLE
client: fix wrong for check

### LINKED ISSUE #7613: lightning or add index failed with '503 Service Unavailable' when inject some pd fault due to PD HTTP client be introduced
### What did you do?
1、run lightning
2、inject some fault to pd leader，or pd rolling restart

### What did you expect to see?
lightning can success

### What did you see instead?
lightning failed

### What version of PD are you using (`pd-server -V`)?
./pd-server -V
 Release Version: v7.6.0-alpha
Edition: Community
Git Commit Hash: 74ef91d1c8a28c1c7d45867bbcd34dcc9ae00cb5
Git Branch: heads/refs/tags/v7.6.0-alpha
UTC Build Time:  2023-12-21 11:38:36
2023-12-23T04:16:24.558+0800

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Close #7613

### What is changed and how does it work?

```commit-message
fix wrong for loop check
```

Due to https://github.com/tikv/pd/blob/master/client/http/client.go#L164 check, when leaderIdx is 0, will not enter loop

### Check List

Tests

- Integration test
- Manual test (add detailed scripts or steps below)

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/http/client.go [hunk 1]
```diff
@@ -161,7 +161,10 @@ func (ci *clientInner) requestWithRetry(
 			zap.String("source", ci.source), zap.Int("leader-idx", leaderAddrIdx), zap.String("addr", addr), zap.Error(err))
 	}
 	// Try to send the request to the other PD followers.
-	for idx := 0; idx < len(pdAddrs) && idx != leaderAddrIdx; idx++ {
+	for idx := 0; idx < len(pdAddrs); idx++ {
```
Comment (rleungx) [inline client/http/client.go:164]: nice catch

**Review by rleungx — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

**Comment by HuSharp** [comment]
test seems to check

File: tests/integrations/client/http_client_test.go [hunk 2]
```diff
@@ -49,7 +51,7 @@ func (suite *httpClientTestSuite) SetupSuite() {
 	re := suite.Require()
 	var err error
 	suite.ctx, suite.cancelFunc = context.WithCancel(context.Background())
-	suite.cluster, err = tests.NewTestCluster(suite.ctx, 2)
+	suite.cluster, err = tests.NewTestCluster(suite.ctx, 3)
```
Comment (CabinfeverB) [inline tests/integrations/client/http_client_test.go:54]: It affects TestTransferLeader?

Comment (HuSharp) [reply tests/integrations/client/http_client_test.go]: repalced with another test, PTAL!

File: client/http/client_test.go [hunk 3]
```diff
@@ -93,3 +98,74 @@ func TestCallerID(t *testing.T) {
 	c.WithCallerID(expectedVal.Load()).GetRegions(context.Background())
 	c.Close()
 }
+
+func TestRedirectWithMetrics(t *testing.T) {
```
Comment (HuSharp) [inline client/http/client_test.go:102]: @rleungx @CabinfeverB PTAL, thx!

**Review by CabinfeverB — APPROVED** [review]

**Review by rleungx — APPROVED** [review]

**Pull request merged**